lib/IRGen/IRGenModule.cpp

```diff
@@ -781,18 +781,37 @@ static StringRef encodeForceLoadSymbolName(llvm::SmallVectorImpl<char> &buf,
   return os.str();
 }
 
+class TargetLinkerOptionGenerator {
+  static std::string Quoted(StringRef Name) {
+    if (Name.find(' ') == StringRef::npos)
+      return Name.str();
+    return "\"" + Name.str() + "\"";
+  }
+
+  const llvm::Triple T;
+
+public:
+  TargetLinkerOptionGenerator(const std::string &Triple) : T(Triple) {}
+  std::string encode(StringRef Library) const {
+    if (T.isWindowsMSVCEnvironment() || T.isWindowsItaniumEnvironment())
+      return "/DEFAULTLIB:" +
+             Quoted(Library.str() +
+                    (Library.endswith_lower(".lib") ? "" : ".lib"));
+    else if (T.isPS4())
+      return "\01" + Quoted(Library.str());
+    return "-l" + Library.str();
+  }
+};
+
 void IRGenModule::addLinkLibrary(const LinkLibrary &linkLib) {
   llvm::LLVMContext &ctx = Module.getContext();
 
   switch (linkLib.getKind()) {
   case LibraryKind::Library: {
-    // FIXME: Use target-independent linker option.
-    // Clang uses CGM.getTargetCodeGenInfo().getDependentLibraryOption(...).
-    llvm::SmallString<32> buf;
-    buf += "-l";
-    buf += linkLib.getName();
-    auto flag = llvm::MDString::get(ctx, buf);
-    AutolinkEntries.push_back(llvm::MDNode::get(ctx, flag));
+    TargetLinkerOptionGenerator LOG(Module.getTargetTriple());
```

Comment:
I don't see how to get that?  I don't see a `llvm::Triple` being stored anywhere in `LLVMContext`.